distributed/core.py

```diff
@@ -583,11 +726,7 @@ async def handle_comm(self, comm):
                         else:
                             result = handler(**msg)
                         if inspect.iscoroutine(result):
-                            result = asyncio.create_task(
-                                result, name=f"handle-comm-{address}-{op}"
-                            )
-                            self._ongoing_coroutines.add(result)
-                            result.add_done_callback(self._ongoing_coroutines.remove)
+                            result = self._ongoing_comm_handlers.schedule(result)
                             result = await result
```

Comment:
this can result in a `TypeError: object NoneType can't be used in 'await' expression`